Manipulators/StandardManipulator.cs

```diff
@@ -67,7 +67,16 @@ void Update()
 
 		void OnTranslateDragging(BaseHandle handle, HandleEventData eventData)
 		{
-			translate(eventData.deltaPosition, eventData.rayOrigin, !(handle is SphereHandle));
+			ConstrainedAxis constraints = 0;
+			var linear = handle as LinearHandle;
```

Comment:
Create an internal interface called IAxisConstraints that exposes the property, so you can simply check for that to get the constraints off instead of checking for specific classes.